app/controllers/api/base_controller/logger.rb

```diff
@@ -94,6 +94,7 @@ def log_request_body
       end
 
       def log_request(header, data)
+        data = data.merge(:request_id => request.request_id) if data.respond_to?(:merge)
```

Comment:
is request_id actually sent in the request payload somewhere or are you only logging it? (I'm only concerned that it will cause confusion since this is logging the actual request.)

Reply:
It's derived from the header or from the rack middleware.

> Returns the unique request id, which is based on either the X-Request-Id header that can be generated by a firewall, load balancer, or web server, or by the [RequestId](https://api.rubyonrails.org/classes/ActionDispatch/RequestId.html) middleware (which sets the action_dispatch.request_id environment variable).

>This unique ID is useful for tracing a request from end-to-end as part of logging or debugging. This relies on the Rack variable set by the [ActionDispatch::RequestId](https://api.rubyonrails.org/classes/ActionDispatch/RequestId.html) middleware.

From: https://api.rubyonrails.org/classes/ActionDispatch/Request.html#method-i-request_id

Reply:
Note, this isn't injecting anything into the request.  It's capturing an additional thing to log from the original request.  It's for logging purposes only.